project-management/src/main/java/life/qbic/projectmanagement/application/ontology/TerminologyService.java

```diff
@@ -0,0 +1,98 @@
+package life.qbic.projectmanagement.application.ontology;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import life.qbic.application.commons.ApplicationException;
+import life.qbic.application.commons.ApplicationException.ErrorCode;
+import life.qbic.application.commons.ApplicationException.ErrorParameters;
+import life.qbic.projectmanagement.domain.model.OntologyTerm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * <b>Terminology Service</b>
+ *
+ * <p>Alternative terminology lookup that is supposed to integrate external resource.</p>
+ *
+ * @since 1.4.0
+ */
+@Service
+public class TerminologyService {
+
+  private final TerminologySelect terminologySelect;
+
+  @Autowired
+  public TerminologyService(TerminologySelect terminologySelect) {
+    this.terminologySelect = Objects.requireNonNull(terminologySelect);
+  }
+
+  /**
+   * Queries possible matching ontology terms given the provided search term. This method should be
+   * used by clients that want to support an auto-suggest use case. The implementation needs to
+   * optimise for fast lookup times.
+   * <p>
+   * <b>NOTE!</b>
+   * <p>
+   * The resulting {@link OntologyTerm} objects are not guaranteed to contain values for properties
+   * like "description". This depends highly on the implementation, use the
+   * {@link TerminologySelect#search(String, int, int)} method for a rich search.
+   *
+   * @param searchTerm a full or partial term that the client wants to select from potentials search
+   *                   hits
+   * @param offset     0 for starting the listing from the beginning of all possible matches, or
+   *                   slice through the results with an offset
+   * @param limit      the maximum number of matches returned per search
+   * @return a list of matching terms given the provided search term
+   * @since 1.4.0
+   */
+  public List<OntologyTerm> query(String searchTerm, int offset, int limit)
+      throws ApplicationException {
+    try {
+      return terminologySelect.query(searchTerm, offset, limit).stream().map(OntologyTerm::from)
+          .toList();
+    } catch (LookupException e) {
+      throw new ApplicationException(ErrorCode.SERVICE_FAILED, ErrorParameters.empty());
```

Comment:
No need to swallow the stack trace. Please set the cause of your exception accordingly. You can rely on the global exception handler to log your exception and prevent anything from reaching the users.
The message in the application exception can be used for debug purposes and provide information about the current state. It will not be presented to the user.
```suggestion
      throw new ApplicationException("Lookup for %s failed.".formatted(searchTerm), e, ErrorCode.SERVICE_FAILED, ErrorParameters.empty());

```